pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -172,14 +174,367 @@ public static String javaPrimitiveToObjectType(String primitiveFieldType) {
 		};
 	}
 
+	/**
+	 * Recursively calculates the hashcode for a message fields.
+	 *
+	 * @param fields The fields of this object.
+	 * @param generatedCodeSoFar The accumulated hash code so far.
+	 * @return The generated code for getting the hashCode value.
+	 */
+	public static String getFieldsHashCode(final List<Field> fields, String generatedCodeSoFar) {
+		for(Field f : fields) {
+			if (f.parent() != null) {
+				final OneOfField oneOfField = f.parent();
+				generatedCodeSoFar += getFieldsHashCode(oneOfField.fields(), generatedCodeSoFar);
+			}
+
+			if (f.optionalValueType()) {
+				generatedCodeSoFar = getOptionalHashCodeGeneration(generatedCodeSoFar, f);
+			}
+			else if (f.repeated()) {
+				generatedCodeSoFar = getRepeatedHashCodeGeneration(generatedCodeSoFar, f);
+			}
+			else if (f.nameCamelFirstLower() != null) {
+				if (f.type() == Field.FieldType.FIXED32 ||
+						f.type() == Field.FieldType.INT32 ||
+						f.type() == Field.FieldType.SFIXED32 ||
+						f.type() == Field.FieldType.SINT32 ||
+						f.type() == Field.FieldType.UINT32) {
+					generatedCodeSoFar += (
+               		"""
+					if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + Integer.hashCode($fieldName);
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.FIXED64 ||
+						f.type() == Field.FieldType.INT64 ||
+						f.type() == Field.FieldType.SFIXED64 ||
+						f.type() == Field.FieldType.SINT64 ||
+						f.type() == Field.FieldType.UINT64) {
+					generatedCodeSoFar += (
+               		"""
+					if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + Long.hashCode($fieldName);
```

Comment:
Should the multiplier be 31 for all field types, or something different?

Reply:
As far as I can tell, the usage of 31 is universal in this particular context. As it turned out there is a good explanation for that:

Computational Efficiency: The choice of 31 can be attributed to the fact that it allows for an efficient implementation of multiplication. In binary form, 31 is 11111, and multiplying by 31 can be efficiently implemented as a shift and a subtraction: (x << 5) - x. This is much faster than a general multiplication operation.

Balancing Speed and Distribution: While there are other primes, 31 offers a good balance between computational efficiency (due to its binary representation) and the effectiveness in producing distinct hash codes for different inputs.